uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-requests_test.js

```diff
@@ -1,30 +1,899 @@
     'use strict';
 
     describe('managerLeaveReport', function () {
-      var $compile, $log, $rootScope, component, controller;
+      var $compile,
+        $log,
+        $q,
+        $provide,
+        $rootScope,
+        component,
+        controller,
+        OptionGroup,
+        AbsenceType,
+        AbsencePeriod,
+        LeaveRequest,
+        Contact,
+        ContactAPIMock,
+        OptionGroupAPIMock;
 
-      beforeEach(module('leave-absences.templates', 'manager-leave'));
-      beforeEach(inject(function (_$compile_, _$log_, _$rootScope_) {
+      beforeEach(module('leave-absences.templates', 'manager-leave', 'leave-absences.mocks', function (_$provide_) {
+        $provide = _$provide_;
+      }));
+
+      beforeEach(inject(function (AbsencePeriodAPIMock, AbsenceTypeAPIMock, LeaveRequestAPIMock) {
+        $provide.value('AbsencePeriodAPI', AbsencePeriodAPIMock);
+        $provide.value('AbsenceTypeAPI', AbsenceTypeAPIMock);
+        $provide.value('LeaveRequestAPI', LeaveRequestAPIMock);
+      }));
+
+      beforeEach(inject(['api.contact.mock', function (_ContactAPIMock_) {
+        ContactAPIMock = _ContactAPIMock_;
+      }]));
+
+      beforeEach(inject(function (_$compile_, _$log_, _$rootScope_, _$q_, _OptionGroup_, _OptionGroupAPIMock_,
+                                  _AbsencePeriod_, _AbsenceType_, _LeaveRequest_, _Contact_) {
         $compile = _$compile_;
         $log = _$log_;
+        $q = _$q_;
         $rootScope = _$rootScope_;
+        OptionGroupAPIMock = _OptionGroupAPIMock_;
+        OptionGroup = _OptionGroup_;
+        AbsenceType = _AbsenceType_;
+        AbsencePeriod = _AbsencePeriod_;
+        LeaveRequest = _LeaveRequest_;
+        Contact = _Contact_
+      }));
 
+      beforeEach(function () {
         spyOn($log, 'debug');
 
+        spyOn(AbsencePeriod, 'all').and.callThrough();
+        spyOn(AbsenceType, 'all').and.callThrough();
+
+        spyOn(Contact, 'all').and.callFake(function () {
+          return $q.resolve(ContactAPIMock.mockedContacts());
+        });
+
+        spyOn(OptionGroup, 'valuesOf').and.callFake(function (name) {
+          return OptionGroupAPIMock.valuesOf(name);
+        });
+      });
+
+      beforeEach(function () {
         compileComponent();
-      }));
+      });
 
       it('is initialized', function () {
         expect($log.debug).toHaveBeenCalled();
       });
 
+      describe('initially', function () {
+
+        it('the filter section is closed', function () {
+          expect(controller.isFilterExpanded).toBe(false);
+        });
+
+        describe('pagination', function () {
+
+          it('is set to page 1', function () {
+            expect(controller.pagination.page).toBe(1);
+          });
+
+          it('has page size set to 7', function () {
+            expect(controller.pagination.size).toBe(7);
+          });
+        });
+      });
+
+      describe('data loading::', function () {
```

Comment:
Actually i see that you use it somehow to mark different sections? Not sure though. Can't you just use plain english?